File: Projects/Framework-Samples/Csla/VB/PetShop/Source/PetShop.UI/Controls/ItemsControl.ascx.vb
Imports PetShop.Business

Partial Public Class ItemsControl
    Inherits System.Web.UI.UserControl

    ''' <summary>
    ''' Rebind control
    ''' </summary>
    Protected Sub PageChanged(ByVal sender As Object, ByVal e As DataGridPageChangedEventArgs)
        'reset index
        itemsGrid.CurrentPageIndex = e.NewPageIndex

        'get category id
        Dim productId As String = Request.QueryString("productId")

        itemsGrid.DataSource = Product.GetProduct(productId).Items
        itemsGrid.DataBind()
    End Sub

End Class

File: tags/CSLA-4.0-Samples/CSharp/PetShop/Source/PetShop.UI/Controls/ItemsControl.ascx.cs
using System;
using System.Web.UI;
using System.Web.UI.WebControls;

using PetShop.Business;

namespace PetShop.UI.Controls
{
    public partial class ItemsControl : UserControl
    {
        /// <summary>
        /// Rebind control 
        /// </summary>
        protected void PageChanged(object sender, DataGridPageChangedEventArgs e)
        {
            //reset index
            itemsGrid.CurrentPageIndex = e.NewPageIndex;

            //get category id
            string productId = Request.QueryString["productId"];

            itemsGrid.DataSource = Product.GetByProductId(productId).Items;
            itemsGrid.DataBind();
        }
    }
}